src/Easify.Exports.IntegrationTests/CsvFileGenerationWithServiceCollectionTests.cs

```diff
@@ -0,0 +1,244 @@
+
+            actual.Should().NotBe(null);
+        }
+
+        [Fact]
+        public async Task Should_ExportAsync_UploadTheFileInAzureBlobStorageWithSharedKey()
+        {
+            var configPath = Path.Combine(Path.GetDirectoryName(GetType().Assembly.Location) ?? string.Empty,
+                "appsettings.json");
+            var entities = _fixture.FakeEntityList<SampleEntity>(5);
+            var configuration = new ConfigurationBuilder().AddJsonFile(configPath, false).Build();
+            var storageTargets = new[]
+            {
+                new StorageTarget
+                {
+                    TargetLocation = "mezz/valuations/automated/", StorageTargetType = StorageTargetType.BlobStorage
+                }
+            };
+            var services = new ServiceCollection();
+            services.AddLogging();
+            services.AddAccountWithSharedKeyStorage(configuration);
+            services.AddCsv(c => { c.Register<SampleEntityMap, SampleEntity>(); });
+            services.AddTransient(sp => _fixture.Fake<ICsvFileWriter>());
+
+            var serviceProvider = services.BuildServiceProvider();
+            var sut = serviceProvider.GetRequiredService<IFileExporter>();
+
+            var actual = await sut.ExportAsync(entities, new ExporterOptions(DateTime.Today, storageTargets));
+
+            actual.Should().NotBe(null);
+            actual.HasError.Should().BeFalse();
+        }
+
+        [Fact]
+        public async Task Should_ExportAsync_UploadTheFileInAzureBlobStorageWithAzureAd()
+        {
+            var configPath = Path.Combine(Path.GetDirectoryName(GetType().Assembly.Location) ?? string.Empty,
+                "appsettings.json");
+            var entities = _fixture.FakeEntityList<SampleEntity>(5);
+            var configuration = new ConfigurationBuilder().AddJsonFile(configPath, false).Build();
+            var storageTargets = new[]
+            {
+                new StorageTarget
+                {
+                    TargetLocation = "mezz/valuations/manual/", StorageTargetType = StorageTargetType.BlobStorage
+                }
+            };
+            var services = new ServiceCollection();
+            services.AddLogging();
+            services.AddAccountWithAzureAdStorage(configuration);
+            services.AddCsv(c => { c.Register<SampleEntityMap, SampleEntity>(); });
+            services.AddTransient(sp => _fixture.Fake<ICsvFileWriter>());
+
+            var serviceProvider = services.BuildServiceProvider();
+            var sut = serviceProvider.GetRequiredService<IFileExporter>();
+
+            var actual = await sut.ExportAsync(entities, new ExporterOptions(DateTime.Today, storageTargets));
+
+            actual.Should().NotBe(null);
+            actual.HasError.Should().BeFalse();
+        }
+
+        //[Fact]
+        public async Task Should_ExportAsync_UploadTheFileInAzureBlobStorageWithEmulator()
```

Comment:
Same as above

Reply:
Cannot be running with no access to AWS. Needs to find another way